base/exports.jl

```diff
@@ -1149,6 +1149,7 @@ export
     ntoh,
     open,
     pipe,
+    Pipe,
```

Comment:
Looking at the test case below:

```
p = Pipe()
run(pipe(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr = p))
@test readall(p) == "Hello World\n"
```

I definitely like this API. I agree it's confusing to have both pipe and Pipe though. Is there something we can rename `Pipe` to? FIFO? PipeStream? IOPipe?